components/script/script_thread.rs

```diff
@@ -325,9 +325,15 @@ pub struct ScriptThread {
     /// Unminify Css.
     unminify_css: bool,
 
-    /// User content manager
+    /// The [`UserContents`] for each `WebView` hosted by this `ScriptThread`.
     #[no_trace]
-    user_content_manager: UserContentManager,
+    user_contents_for_webview: RefCell<FxHashMap<WebViewId, Rc<UserContents>>>,
+
+    // The default `UserContents` used when embedder has not set a `UserContentManager`
+    // on a `WebView`. This must be empty and should have no effect on the pages
+    // loaded in the `WebView`.
+    #[no_trace]
+    default_user_contents: OnceCell<Rc<UserContents>>,
```

Comment:
Instead of doing this, I think it makes sense that the `Window` stores and `Option`, if it is feasible. I think that's a more direct way of representing this situation.

Reply:
I had it this way originally, but it seemed like this case would need to be handled for other types of user content as well, like stylesheets in the future and it seemed easier to assume we always default to empty. I've removed this logic now and switched back to an `Option` in `Window` .